jgrapht-core/src/main/java/org/jgrapht/generate/DirectedScaleFreeGraphGenerator.java

```diff
@@ -0,0 +1,376 @@
+     * @param deltaOut The out-degree bias used for Beta and Gamma.
+     * @param targetEdges Target total number of edges to reach. It has a higher priority than
+     *        {@link #targetNodes}. Zero is a valid value.<br>
+     *        If negative number, the user does not care about the total number of edges and is
+     *        interested only in the number of nodes, therefore, {@link #targetNodes} will be
+     *        considered instead. Otherwise, {@link #targetEdges} will be considered and
+     *        {@link #targetEdges} will be ignored.
+     * @param targetNodes Target number of nodes to reach. Zero is a valid value.<br>
+     *        This parameter has lower priority than {@link #targetEdges} and will be used only if
+     *        {@link #targetEdges} given is a <i>negative</i> number.
+     * @param rng The {@link Random} object to use.
+     */
+    public DirectedScaleFreeGraphGenerator(
+        float alpha, float gamma, float deltaIn, float deltaOut, int targetEdges, int targetNodes,
+        Random rng)
+    {
+        this.alpha = alpha;
+        // this.gamma = gamma;
+        this.alphaPlusBeta = 1.0f - gamma;
+        this.deltaIn = deltaIn;
+        this.deltaOut = deltaOut;
+        this.targetEdges = targetEdges;
+        this.targetNodes = targetNodes;
+        this.rng = Objects.requireNonNull(rng, "Random number generator cannot be null");
+
+        // Do several checks on the parameters
+        if (alpha < 0 || gamma < 0 || alpha + gamma > 1) {
+            throw new IllegalArgumentException(
+                String.format("alpha and gamma values of (%f, %f) are invalid", alpha, gamma));
+        }
+        if (deltaIn < 0 || deltaOut < 0) {
+            throw new IllegalArgumentException(
+                String
+                    .format(
+                        "deltaIn and deltaOut values of (%f, %f) are invalid", deltaIn, deltaOut));
+        }
+        if (targetEdges < 0) {
+            // priority is for targetNodes
+            if (targetNodes < 0) {
+                throw new IllegalArgumentException(
+                    "can not have both targetEdges and targetNodes not set.");
+            }
+        }
+
+    }
+
+    /**
+     * Generates an instance.
+     * 
+     * @param target the target graph
+     * @param resultMap not used by this generator, can be null
+     */
+    @Override
+    public void generateGraph(Graph<V, E> target, Map<String, V> resultMap)
+    {
+        Set<V> newNodesSet = new HashSet<>();
+        Set<E> newEdgesSet = new HashSet<>();
+        final int MAX_VERTEX_FAILURES = 1000;
+
+        if(targetEdges == 0 || (targetEdges < 0 && targetNodes == 0))
+            return;
+        V initV = target.addVertex();
+        newNodesSet.add(initV);
+
+        int failuresCounter = 0;
+        // grow network now
+        while (targetEdges >= 0 ? targetEdges > newEdgesSet.size()
+            : newNodesSet.size() <= targetNodes)
```

Comment:
No. This is on purpose.
Because the graph may continue to grow (in edges), after it reached the maximum number of nodes.

In this I allow the graph to add more internal edges in case the tributary is \beta, but not \alpha nor \gamma.

I will document that clearly.